src/main/java/net/dv8tion/jda/core/entities/MessageChannel.java

```diff
@@ -545,30 +549,40 @@ protected void handleResponse(Response response, Request<Message> request)
      */
     default RestAction<Message> sendFile(InputStream data, String fileName, Message message)
     {
-        Args.notNull(data, "data InputStream");
-        Args.notNull(fileName, "fileName");
+        Checks.notNull(data, "data InputStream");
+        Checks.notNull(fileName, "fileName");
 
         Route.CompiledRoute route = Route.Messages.SEND_MESSAGE.compile(getId());
-        MultipartBody body = Unirest.post(Requester.DISCORD_API_PREFIX + route.getCompiledRoute())
-                .fields(null); //We use this to change from an HttpRequest to a MultipartBody
-
-        body.field("file", data, fileName);
+        MultipartBody.Builder builder = new okhttp3.MultipartBody.Builder()
+                .setType(MultipartBody.FORM);
+        try
+        {
+            byte[] bytes = IOUtil.readFully(data);
```

Comment:
https://github.com/DV8FromTheWorld/JDA/blob/experimental/message-rw/src/main/java/net/dv8tion/jda/core/entities/Message.java#L977-L1006 this can easily be made public in `experimental/message-rw` later on.